### PR TITLE
[ONNX] Convert index_put with bool inputs to masked_scatter

### LINKED ISSUE #34054: Export to ONNX models including logical masks
## 🐛 Bug

This is a bug in torch.onnx module during export. Unable to handle with a logical mask due to TypeError.

## To Reproduce

1. type to the forward method something like
> mask = input_mask.clone()
> mask[mask != some_constant] = 1
> mask[mask == some_constant] = 0
2. export the model to onnx: 
>torch.onnx.export(model, dummy_input, filename, verbose=True, opset_version=11)

3. load the model by 
>onnxruntime.InferenceSession(filename)

4. catch the error. 
>This is an invalid model. Type Error: Type 'tensor(bool)' of input parameter (337) of operator (ScatterND) in node () is invalid.

5. see the log
>...
  %334 : Bool(20, 50, 6) = onnx::Cast[to=9](%333) # C:\Users\....\file.py:49:0
  %335 : Long() = onnx::Constant[value={1}]() # C:\Users\....\file.py:49:0
  %336 : Tensor = onnx::Shape(%334)
  %337 : Tensor = onnx::Unsqueeze[axes=[-1]](%334)
  %338 : Tensor = onnx::Shape(%atom_degree_list)
  %339 : Tensor = onnx::Constant[value={0}]()
  %340 : Tensor = onnx::Constant[value={1}]()
  %341 : Tensor = onnx::Constant[value={9223372036854775807}]()
  %342 : Tensor = onnx::Slice(%338, %340, %341, %339)
  %343 : Tensor = onnx::Concat[axis=0](%336, %342)
  %344 : Tensor = onnx::Reshape(%335, %343)
  %345 : Long(20, 50, 6) = onnx::ScatterND(%input_mask, %337, %344) # C:\Users\....\file.py:49:0
...

## Expected behavior
The workaround:
I can alter the code in the forward method like 
> mask = input_mask.clone()
> mask = mask - some_constant
> mask = numpy.heaviside(mask.abs(), 0).long()

and load the exported model successfully by onnxruntime. 
But I'm trying to export a network from a third-party library. Therefore I'm not allowed to alter the code.

## Environment

 - PyTorch Version : 1.4.0
 - OS: Windows 10 Pro 64 bit
 - How you installed PyTorch (`conda`, `pip`, source): current repo (git clone ...)
 - Build command you used (if compiling from source): setup.py build/install
 - Python version: 3.6
 - CUDA/cuDNN version: None
 - GPU models and configuration: None
 - Any other relevant information: opset 11

cc @houseroad @spandantiwari @lara-hdr @BowenBao @neginraoof

### PR DESCRIPTION
**Pull request by shubhambhokare1**

### CODE REVIEW

**Comment by BowenBao** [comment]
Can you add the case to fix in the unit test? Also remember to resolve the clang-format issue.

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 1]
```diff
@@ -104,10 +104,47 @@ static void FuseWithListUnpack(Block* b) {
...
+  TORCH_INTERNAL_ASSERT(n->inputs().size() == 4);
+  TORCH_INTERNAL_ASSERT(n->outputs().size() == 1);
+
+  auto index_put_node = n;
+  auto graph = index_put_node->owningGraph();
+
+  Value* orig_data = index_put_node->input(0);
+  auto new_masked_scatter = graph->insert(
+      aten::masked_scatter,
+      {orig_data,
```
Comment (KsenijaS) [inline torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:119]: why create orig_data if it's only used here?

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 2]
```diff
@@ -104,10 +104,47 @@ static void FuseWithListUnpack(Block* b) {
   }
 }
 
+void BoolIndexPutToMaskedScatter(Node* n){
```
Comment (KsenijaS) [inline torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:107]: are you going to add tests for this change?

File: torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp [hunk 3]
```diff
@@ -104,10 +104,47 @@ static void FuseWithListUnpack(Block* b) {
...
+  std::cout << n->inputs() << std::endl;
+  std::cout << n->outputs() << std::endl;
```
Comment (KsenijaS) [inline torch/csrc/jit/passes/onnx/preprocess_for_onnx.cpp:109]: why are inputs and outputs printed?

**Pull request closed without merging**